### PR TITLE
gradually hide fileList elements as width shrinks

### PR DESCRIPTION
**Pull request by felixheidecke**
The effort to make the filesApp responsive.

TODO:

- [x] correct behavior when the sidebar is closed
- [x] correct behavior when the sidebar is open
- [ ] use the correct icon (dots) for the collapsed actions-button

### Tablet

files_app_tablet_landscape

FireShot Capture 011 - ownCloud Phoenix - localhost

### Tablet (portrait)

files_app_tablet_portrait

FireShot Capture 012 - ownCloud Phoenix - localhost

### Mobile

files_app_mobile

FireShot Capture 014 - ownCloud Phoenix - localhost

**Landscape**

FireShot Capture 013 - ownCloud Phoenix - localhost

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@felixheidecke ready for review or still developing? Label indicates the later ....

**Comment by DeepDiver1975** [comment]
the actions button on small screens is not yet working - so: still developing .... 

anyway - looks much better already - THX

**Comment by felixheidecke** [comment]
@DeepDiver1975 yes … still in dev, as I also need proper behavior with the sidebar open. The dots icon will be a part of the ODS.

**Comment by felixheidecke** [comment]
I'm not sure, how much space in the sidebar is needed, so I just slapped on medium and large. Let's see if it suffices … rework if it doesn't.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFavorites</strong> failed. Please find the screenshots inside ...</summary><p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFolders</strong> failed. Please find the screenshots inside ...</summary><p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIUpload</strong> failed. Please find the screenshots inside ...</summary><p>














</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary><p>


















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




























</p></details>

File: apps/files/src/components/FilesApp.vue [hunk 1]
```diff
@@ -1,14 +1,14 @@
   <template>
     <div class="oc-app" id="files-app" @dragover="$_ocApp_dragOver">
-      <oc-app-layout :rightHidden="selectedFiles.length === 0">
-        <template slot="center">
-          <oc-loader id="files-list-progress" v-if="loadingFolder"></oc-loader>
+      <oc-grid>
+        <div class="uk-width-expand" :class="{ 'uk-visible@s' : _sidebarOpen }">
+          <oc-loader id="files-list-progress" v-if="loading"></oc-loader>
```
Comment (DeepDiver1975) [inline apps/files/src/components/FilesApp.vue]:
```
[Vue warn]: Property or method "loading" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

found in

---> <FilesApp> at src/components/FilesApp.vue
       <Phoenix> at src/Phoenix.vue
         <Root>

File: apps/files/src/components/FilesApp.vue [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
           <file-list @toggle="toggleFileSelect" @FileAction="openFileActionBar" :fileData="activeFiles" @sideBarOpen="openSideBar"/>
-        </template>
-        <template slot="right">
-          <file-details :items="selectedFiles" :starsEnabled="false" :checkboxEnabled="false" ref="fileDetails" @reload="$_ocFilesFolder_getFolder" @reset="resetFileSelection"/>
-        </template>
-      </oc-app-layout>
+        </div>
+        <div class="uk-width-1-1 uk-width-medium@s uk-width-large@l" v-show="_sidebarOpen">
+          <file-details :items="selectedFiles" :starsEnabled="false" :checkboxEnabled="false" ref="fileDetails" @reload="getFolder" @reset="resetFileSelection"/>
```
Comment (DeepDiver1975) [inline apps/files/src/components/FilesApp.vue]:
```
Vue warn]: Property or method "getFolder" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

found in

---> <FilesApp> at src/components/FilesApp.vue
       <Phoenix> at src/Phoenix.vue
         <Root>

File: package.json [hunk 3]
```diff
@@ -83,5 +83,13 @@
     "webpack-copy-plugin": "^0.0.4",
     "webpack-dev-server": "^3.3.1",
     "webpack-merge": "^4.2.1"
+  },
+  "dependencies": {
```
Comment (DeepDiver1975) [inline package.json]:
```
warning package.json: "dependencies" has dependency "owncloud-design-system" with range "^1.0.0-605" that collides with a dependency in "devDependencies" of the same name with version "^1.0.0-611"
warning package.json: "dependencies" has dependency "owncloud-sdk" with range "1.0.0-92" that collides with a dependency in "devDependencies" of the same name with version "1.0.0-107"
warning owncloud_phoenix@0.1.0: "dependencies" has dependency "owncloud-design-system" with range "^1.0.0-605" that collides with a dependency in "devDependencies" of the same name with version "^1.0.0-611"
warning owncloud_phoenix@0.1.0: "dependencies" has dependency "owncloud-sdk" with range "1.0.0-92" that collides with a dependency in "devDependencies" of the same name with version "1.0.0-107"
[1/4] Resolving packages...

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFavorites</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary><p>




</p></details>

**Pull request merged**